py3status/modules/gpmdp.py

```diff
@@ -3,60 +3,148 @@
+    cache_timeout: refresh interval for this module (default 5)
+    format: display format for this module
+        (default '[\?if=is_started [\?if=title [\?if=playing&color=playing
+        {title} - {artist}|\?color=paused (Paused) {title} - {artist}]|
+        \?color=stopped Google Play Music]]')
+    sleep_timeout: sleep interval for this module (default 20)
+
+Control placeholders:
+    {is_started} a boolean based on pgrep data
+    {playing} a boolean based on data status
+    {rating_disliked} a boolean based on data status
+    {rating_liked} a boolean based on data status
 
 Format placeholders:
-    {info}            Print info about now playing song
-    {title}           Print current song title
-    {artist}          Print current song artist
-    {album}           Print current song album
-    {album_art}       Print current song album art URL
-    {time_current}    Print current song time in milliseconds
-    {time_total}      Print total song time in milliseconds
-    {status}          Print whether GPMDP is paused or playing
-    {current}         Print now playing song in "artist - song" format
+    {repeat} song repeat, eg NO_REPEAT, LIST_REPEAT, SINGLE_REPEAT
+    {shuffle} song shuffle, eg NO_SHUFFLE, ALL_SHUFFLE
+    {songLyrics} 100~ lines lyrics
+    {song_albumArt} album art url
+    {song_album} album name
+    {song_artist} artist name
+    {song_title}  title name
+    {time_currenttime} current time in [HH:]MM:SS
+    {time_current} current time in milliseconds
+    {time_totaltime} total time in [HH:]MM:SS
+    {time_total} total time in milliseconds
+    {volume} volume percent
 
+Color options:
+    color_paused: Paused, defaults to color_degraded
+    color_playing: Playing, defaults to color_good
+    color_stopped: Not playing, defaults to color_bad
 
 Requires:
-    gpmdp: http://www.googleplaymusicdesktopplayer.com/
-    gpmdp-remote: https://github.com/iandrewt/gpmdp-remote
+    gpmdp: A beautiful cross platform Desktop Player for Google Play Music
 
-@author Aaron Fields https://twitter.com/spirotot
+@author Aaron Fields https://twitter.com/spirotot, lasers
 @license BSD
 
 SAMPLE OUTPUT
-{'full_text': '♫ Now Playing: The Show Goes On by Lupe Fiasco'}
+{'color': '#00ff00', 'full_text': 'The Show Goes On by Lupe Fiasco'}
+
+paused
+{'color': '#ffff00', 'full_text': '(Paused) The Show Goes On by Lupe Fiasco'}
+
+stopped
+{'color': '#ff0000', 'full_text': 'Google Play Music'}
 """
 
+from __future__ import division
+import json
+import os
+
+GPMDP = '~/.config/Google Play Music Desktop Player/json_store/playback.json'
+PGREP = ['pgrep', '-f', 'Google Play Music Desktop Player']
+
 
 class Py3status:
     """
     """
     # available configuration parameters
     cache_timeout = 5
-    format = u'♫ {info}'
+    format = '[\?if=is_started [\?if=title [\?if=playing&color=playing ' +\
+        '{title} - {artist}|\?color=paused (Paused) {title} - {artist}]|' +\
+        '\?color=stopped Google Play Music]]'
```

Comment:
I detest the `\` line extension it always reminds me of VB which causes psychotic episodes.  There is a little trick in python where strings will append to each other.  It is also the cause of some of the hardest bugs I've tried to find in python but you can do this

```python
    format = ('[\?if=is_started [\?if=title [\?if=playing&color=playing '
        '{title} - {artist}|\?color=paused (Paused) {title} - {artist}]|'
        '\?color=stopped Google Play Music]]')
```

Reply:
:white_check_mark: